kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue

```diff
@@ -166,9 +167,15 @@
       if (
         props.filterByFacilityId !== null ||
         props.filterByFacilityCanSignUp !== null ||
-        props.filterByOnMyOwnFacility !== null
+        props.filterByOnMyOwnFacility !== null ||
+        props.filterByHasFacilities !== null
       ) {
         apiParams.subset_of_users_device = false;
+
+        if (props.filterByHasFacilities === true) {
+          deviceFilters.push(useDeviceHasFacilitiesFilter());
```

Comment:
This works, although it does add some tech debt because if the prop was used in conjunction with the other facility filter props, we'd be calling the API twice. So ideally the existing `useDeviceFacilityFilter` is updated to have a special parameter supporting this type of filtering. Long term, the idea is to remove the filters and use the filter composables directly in usages, which would likely pose the same problem.

Reply:
`useDeviceFacilityFilter` just seemed fundamentally different from what the new function is trying to do where it is basically saying "give me a facility-shaped thing and I'll find if there is a match" vs "are there _any_ facilities at all?" 

I considered updating the `useDeviceFacilityFilter` w/ a second param like `any=false` to mean like "oh actually we only care about if there are any facilities" but this felt stapling functionality onto something where it doesn't really belong.

I definitely get the issue w/ the tech debt here, though, and it _is_ a bit wonky as-is -- but I'm curious what you mean here:

> the idea is to remove the filter props and use the filter composables directly in usages

Like - would we instead be importing from `useDevices` in, say, the SetupWizard where we use the `<SelectDeviceForm>`, then we call the functions from `useDevices` to setup the filters and pass them directly, rather than working it out based on the props being null vs other?

Reply:
> fundamentally different from what the new function is trying to do where it is basically saying "give me a facility-shaped thing and I'll find if there is a match" vs "are there any facilities at all?"

I get that there's fundamentally different use cases, but IMO the logic isn't different enough. For example, I see it like this:

```js
// has matching facility with ID
let match = f => f.id = id;
facilities.filter(match).length > 0;

// has any
match = () => true;
facilities.filter(match).length > 0;
```

When a request to match a specific facility is applied, the 'any match' scenario becomes implicit, because having no facilities or a least a matching facility is the same 'any match' . So filtering by specific properties is really just a subset of 'any' match. The only important aspect would be to ensure that the logic represents a specific filter as a subset, meaning the specific filters take precedence over any match, which works because the results of that again are the same and do not conflict.

> Like - would we instead be importing from useDevices in, say, the SetupWizard where we use the <SelectDeviceForm>, then we call the functions from useDevices to setup the filters and pass them directly, rather than working it out based on the props being null vs other?

That's the idea-- to let the implementation use the composables and remove the complexity of handling and managing multiple props which could conflict or produce issues when used in conjunction.

Reply:
A compromise would be to keep it separate but just move where it's applied so it's impossible for both the specific filters and the any facility filter from being applied simultaneously. Which again works because of the same reasons